protoc-gen-kotlin/lib/src/jvmTest/kotlin/CodeGeneratorTest.kt

```diff
@@ -0,0 +1,123 @@
+
+        assertEquals(result.exitCode, ExitCode.OK)
+        val message1Clazz = result.classLoader.loadClass("foobar.Message1")
+        val message2Clazz = result.classLoader.loadClass("foobar.Message2")
+        message1Clazz.getDeclaredField("intVal")
+        message2Clazz.getDeclaredField("strVal")
+    }
+
+    @Test
+    fun testOneOf_SameNameField() {
+        val result = compileProto("oneof_same_name.proto")
+
+        assertEquals(result.exitCode, ExitCode.OK)
+        val valueClazz = result.classLoader.loadClass("foobar.Value")
+        valueClazz.classLoader.loadClass("foobar.Value\$Value")
+    }
+
+    companion object {
+        const val PROTOC_PLUGIN_PATH = "../jvm/build/install/protoc-gen-kotlin/bin/protoc-gen-kotlin"
+        const val PROTO_PATH = "src/jvmTest/resources/protos"
+        const val DESCRIPTOR_SET_OUT = "build/tmp/fileDescriptor.protoset"
+        private const val DOWNLOAD_PROTOC_VERSION = "3.10.1"
+
+        val protoc: String = (System.getProperty("protoc.path")?.let { "$it/bin/protoc" } ?: "protoc")
+            .let { protoc ->
+                try {
+                    // try executing the protoc binary, this code throws an exception if it cannot be found
+                    ProcessBuilder(protoc).start()
+                    protoc
+                } catch (e: Exception) {
+                    downloadedProtoc()
+                }
+            }
+
+        private fun downloadedProtoc() = try {
+            val filename = "protoc-$DOWNLOAD_PROTOC_VERSION-${getOS()}-${getArch()}.exe"
+            val protocTemp = File.createTempFile("protoc", "tmp")
+            val protoUrl =
+                URL("https://repo1.maven.org/maven2/com/google/protobuf/protoc/$DOWNLOAD_PROTOC_VERSION/$filename")
+            protocTemp.writeBytes(protoUrl.readBytes())
+            protocTemp.setExecutable(true)
+            protocTemp.absolutePath
+        } catch (e: Exception) {
+            throw Exception("failed to download protoc version $DOWNLOAD_PROTOC_VERSION", e)
+        }
+
+        private fun getOS() = System.getProperty("os.name").toLowerCase().let {
+            when {
+                it.contains("windows") -> "windows"
+                it.contains("mac os x") || it.contains("darwin") || it.contains("osx") -> "osx"
+                else -> "linux"
+            }
+        }
+
+        private fun getArch() = System.getProperty("os.arch").toLowerCase().let {
+            when {
+                it.contains("amd64") || it.contains("x86_64") -> "x86_64"
+                else -> "x86_32"
+            }
+        }
+    }
+
+    private fun getFileDescriptorProto(protoFile: String): List<FileDescriptorProto> {
+        val plugin = File(PROTOC_PLUGIN_PATH + ".bat".takeIf {
+            getOS() == "windows"
+        }.orEmpty())
+
+        val proc = ProcessBuilder(
+            protoc,
+            "--proto_path=$PROTO_PATH",
+            "--plugin=protoc-gen-kotlin=${plugin.absoluteFile}",
```

Comment:
You don't need to pass this option if you're using protoc only to output the descriptor set.